tests/Microsoft.DotNet.Docker.Tests/SdkImageTests.cs

```diff
@@ -121,6 +136,129 @@ public void VerifyPowerShellScenario_NonDefaultUser(ProductImageData imageData)
+            }
+            else
+            {
+                dotnetPath = "Program Files\\dotnet";
+            }
+
+            string powerShellCommand =
+                $"Get-ChildItem -File -Force -Recurse '{dotnetPath}' " +
+                "| Get-FileHash -Algorithm SHA512 " +
+                "| select @{name='Value'; expression={$_.Hash + '  ' +$_.Path}} " +
+                "| select -ExpandProperty Value";
+            string command = $"pwsh -Command \"{powerShellCommand}\"";
+
+            string containerFileList = DockerHelper.Run(
+                image: imageData.GetImage(ImageType, DockerHelper),
+                command: command,
+                name: imageData.GetIdentifier("DotnetFolder"));
+
+            IEnumerable<SdkContentFileInfo> actualDotnetFiles = containerFileList
+                .Replace("\\", "/")
+                .Replace("\r\n", "\n")
+                .Split("\n")
+                .Select(output =>
+                {
+                    string[] outputParts = output.Split("  ");
+                    return new SdkContentFileInfo(outputParts[1], outputParts[0]);
+                })
+                .OrderBy(fileInfo => fileInfo.Path)
+                .ToArray();
+
+            IEnumerable<SdkContentFileInfo> expectedDotnetFiles = await GetSdkContentsAsync(imageData);
+
+            Assert.Equal(expectedDotnetFiles.Count(), actualDotnetFiles.Count());
+
+            // Skip file comparisons for 3.1 until https://github.com/dotnet/sdk/issues/11327 is fixed.
+            if (imageData.Version.Major == 3)
+            {
+                return;
+            }
+
+            int fileCount = expectedDotnetFiles.Count();
+            for (int i = 0; i < fileCount; i++)
+            {
+                Assert.Equal(expectedDotnetFiles.ElementAt(i), actualDotnetFiles.ElementAt(i));
+            }
+        }
+
+        private static IEnumerable<SdkContentFileInfo> EnumerateArchiveContents(string path)
+        {
+            using FileStream fileStream = File.OpenRead(path);
+            using IReader reader = ReaderFactory.Open(fileStream);
+            using TempFolderContext tempFolderContext = FileHelper.UseTempFolder();
+            reader.WriteAllToDirectory(tempFolderContext.Path, new ExtractionOptions() { ExtractFullPath = true });
+
+            foreach (FileInfo file in new DirectoryInfo(tempFolderContext.Path).EnumerateFiles("*", SearchOption.AllDirectories))
+            {
+                using SHA512 sha512 = SHA512.Create();
+                byte[] sha512HashBytes = sha512.ComputeHash(File.ReadAllBytes(file.FullName));
+                string sha512Hash = BitConverter.ToString(sha512HashBytes).Replace("-", String.Empty);
+                yield return new SdkContentFileInfo(
+                    file.FullName.Substring(tempFolderContext.Path.Length), sha512Hash);
+            }
+        }
+
+        private async Task<IEnumerable<SdkContentFileInfo>> GetSdkContentsAsync(ProductImageData imageData)
```

Comment:
This name is a little ambiguous as it doesn't indicate what sdk contents are retrieved.  In the scope of this test, I think the reader should assume the SDK docker image not the archive.
```suggestion
        private async Task<IEnumerable<SdkContentFileInfo>> GetSdkArchiveContentsAsync(ProductImageData imageData)
```